### PR TITLE
feat: add Space component

### PR DESCRIPTION
**Pull request by zhanba**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

-----
[View rendered components/space/demo/size.md](https://github.com/zhanba/ant-design/blob/feat/space/components/space/demo/size.md)
[View rendered components/space/index.en-US.md](https://github.com/zhanba/ant-design/blob/feat/space/components/space/index.en-US.md)
[View rendered components/space/index.zh-CN.md](https://github.com/zhanba/ant-design/blob/feat/space/components/space/index.zh-CN.md)

### CODE REVIEW

File: components/space/demo/size.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+```jsx
+import { Space } from 'antd';
+
+class Demo extends React.Component {
+  render() {
+    return (
+      <div>
+        <p>
+          this is a space, <Space /> size is default.
```
Comment (afc163) [inline components/space/demo/size.md]: 不要这样用，按 https://github.com/ant-design/ant-design/issues/19860 作为包裹组件使用。

File: components/config-provider/__tests__/index.test.js [hunk 2]
```diff
@@ -31,4 +32,13 @@ describe('ConfigProvider', () => {
 
     expect(wrapper.find('Button').text()).toBe('确定');
   });
+
+  it('spaceBaseSize', () => {
+    const wrapper = mount(
+      <ConfigProvider spaceBaseSize={7}>
```
Comment (zombieJ) [inline components/config-provider/__tests__/index.test.js]: 参考 `form` 配置包一层：
```tsx
<ConfigProvider space={{ size: 7 }} />
```


ref: https://next.ant.design/components/config-provider/#header

File: components/space/demo/size.md (hunk 1 above)
Comment (zombieJ) [reply components/space/demo/size.md]: Space 管理子元素添加 margin，没有子元素时它不占空间。

File: components/config-provider/__tests__/index.test.js (hunk 2 above)
Comment (zhanba) [reply components/config-provider/__tests__/index.test.js]: 这里有点不一样，这个baseSize不是Space组件的props，而是用来计算size大小的一个基数

Comment (zombieJ) [reply components/config-provider/__tests__/index.test.js]: 那就叫做 `baseSize`

**Pull request closed without merging**